marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -38,14 +36,39 @@ mkMainFrame =
           }
     }
 
+initialState :: State
+initialState =
+  { overlay: Nothing
+  , screen: Home
+  , card: Nothing
+  , contactState: Contact.initialState
+  , notifications: []
+  , templates: []
+  , contracts: []
+  , on: true
+  }
+
 handleQuery :: forall a m. Query a -> HalogenM State Action ChildSlots Msg m (Maybe a)
 handleQuery (ReceiveWebSocketMessage msg next) = do
   current <- use _on
   modifying _on not
   pure $ Just next
 
 handleAction :: forall m. MonadAff m => Action -> HalogenM State Action ChildSlots Msg m Unit
-handleAction Init = pure unit
+handleAction Init = do
+  mCachedContactsJson <- liftEffect $ getItem contactsLocalStorageKey
+  case mCachedContactsJson of
+    Just json -> do
+      let
+        contacts =
+          runExcept
+            $ do
+                foreignJson <- parseJSON json
+                decode foreignJson
+      case contacts of
+        Right cachedContacts -> assign (_contactState <<< _contacts) cachedContacts
```

Comment:
I think you can move this up into the `ExceptT` so that you have
```purescript
            do
                foreignJson <- parseJSON json
                contacts <- decode foreignJson
                assign (_contactState <<< _contacts) cachedContacts
```